pyscitt/pyscitt/receipt.py

```diff
@@ -21,27 +24,82 @@
 
 HEADER_PARAM_TREE_ALGORITHM = "tree_alg"
 TREE_ALGORITHM_CCF = "CCF"
-
-
-def hdr_as_dict(phdr: dict) -> dict:
+COMMON_CWT_KEYS_MAP = {
+    1: "iss",
+    2: "sub",
+    3: "aud",
+    4: "exp",
+    5: "nbf",
+    6: "iat",
+    7: "cti",
+}
+
+
+def display_cwt_key(item: Any) -> Union[int, str]:
+    """Convert a CWT key to a string for pretty-printing."""
+    out = str(item)
+    return COMMON_CWT_KEYS_MAP.get(item, out)
+
+
+def display_cbor_val(item: Any) -> str:
+    """Convert a CBOR item to a string for pretty-printing."""
+    out = str(item)
+    if hasattr(item, "__name__"):
+        out = item.__name__
+    elif isinstance(item, datetime.datetime):
+        out = item.isoformat()
+    elif type(item) is bytes:
+        out = item.hex()
+    return out
+
+
+def cbor_as_dict(cbor_obj: Any, cbor_obj_key: Any = None) -> Any:
     """
-    Return a representation of a list of COSE header parameters that
-    is amenable to pretty-printing.
+    Return a printable representation of a CBOR object.
```

Comment:
Is there some risk that some of these conversions may fail with an unhandled exception? If so, it could make sense to add a try-catch to handle possible general exceptions gracefully (either here or upstream when this method is called to print the receipt).

Reply:
Added try/catch around receipt parsing, added a default serializer for json serialization to avoid unexpected cbor breaking the command, renamed `cbor_as_dict` to `cbor_to_printable` as it does not only return structs.